x-pack/legacy/plugins/apm/server/routes/default_api_types.ts

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as t from 'io-ts';
+import { either } from 'fp-ts/lib/Either';
+
+const dateRt = new t.Type<string, string, unknown>(
+  'DateAsString',
+  t.string.is,
+  (u, c) =>
+    either.chain(t.string.validate(u, c), s => {
+      const d = new Date(s);
+      return isNaN(d.getTime()) ? t.failure(u, c) : t.success(s);
+    }),
+  a => a
+);
+
+export const rangeRt = t.type({
+  start: dateRt,
+  end: dateRt
+});
+
+export const uiFiltersRt = t.type({ uiFilters: t.string });
+
+export const debugRt = t.partial({ debug: t.boolean });
+
+export const jsonRt = new t.Type<any, string, unknown>(
+  'JSONAsString',
+  t.string.is,
+  (u, c) =>
```

Comment:
`u` and `c`? Is that convention or what do they refer to?

Reply:
u = input, c = context. I'll write those out though, more understandable for people not familiar with `io-ts`.